public/app/features/explore/TraceView/components/TracePageHeader/NewTracePageSearchBar.tsx

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { css } from '@emotion/css';
+import React, { memo, Dispatch, SetStateAction, useEffect } from 'react';
+
+import { config, reportInteraction } from '@grafana/runtime';
+import { Button, useStyles2 } from '@grafana/ui';
+
+import { SearchProps } from '../../useSearch';
+
+export type TracePageSearchBarProps = {
+  search: SearchProps;
+  setSearch: React.Dispatch<React.SetStateAction<SearchProps>>;
+  spanFilterMatches: Set<string> | undefined;
+  focusedSpanIdForSearch: string;
+  setFocusedSpanIdForSearch: Dispatch<SetStateAction<string>>;
+  datasourceType: string;
+};
+
+export default memo(function NewTracePageSearchBar(props: TracePageSearchBarProps) {
+  const { search, spanFilterMatches, focusedSpanIdForSearch, setFocusedSpanIdForSearch, datasourceType } = props;
+  const styles = useStyles2(getStyles);
+
+  useEffect(() => {
+    setFocusedSpanIdForSearch('');
+  }, [search, setFocusedSpanIdForSearch]);
+
+  const nextResult = () => {
+    reportInteraction('grafana_traces_trace_view_find_next_prev_clicked', {
+      datasourceType: datasourceType,
+      grafana_version: config.buildInfo.version,
+      direction: 'next',
+    });
+
+    const spanMatches = Array.from(spanFilterMatches!);
```

Comment:
An empty set `new Set()` would be of type `unknown` so it would be similar no? In this case, next/prev will not be available unless there are matches to cycle through.